WordPress/gutenberg
# Add support for orientation-based block movers to core/social-links

## What?
Currently, when you select a social link the block movers are always vertical arrows. The "up" arrow moves the block to the left, and the "down" arrow moves the block to the right.

Closes #48451.

## Why?
It's confusing that the up icon moves a social link block to the left. 

## How?
Add `orientation` prop to core/social-links `useInnerBlocksProps`, based on the block's `layout.orientation` value.

If `vertical` orientation, the block movers will render up/down icons. If `horizontal` orientation, the block movers will render left/right icons. 

## Testing Instructions
1. Add a Social Links block.
2. Add at least two links within the block.
3. Select a link, see left and right arrow movers that move it left and right.
4. Open block inspector of the parent Social Links block.
5. Set layout orientation to vertical.
6. Select a link, see up and down arrow movers that move it up and down.

## Screenshots or screencast 

Movers render vertically or horizontally, based on layout orientation: 

https://user-images.githubusercontent.com/1813435/221377098-6e5c4234-314f-43b5-9d80-c260420a8be3.mp4
